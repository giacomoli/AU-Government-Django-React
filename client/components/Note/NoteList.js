import React from 'react'

import Note from './Note'

export default ({ notes, matterId, contactId, user }) => {
  if (notes.length > 0) {
    return (
      <table id="mainTable" className="table table-bordered table-striped m-b-0">
        <thead>
          <tr>
            <th>Date time</th>
            <th width="70%">Note</th>
            <th>Staff member</th>
          </tr>
        </thead>
        <tbody>
          {notes.map(note => (
            <Note
              key={note.id}
              matterId={matterId}
              contactId={contactId}
              user={user}
              form={`note${note.id}`}
              note={note}
            />
          ))}
        </tbody>
        <style jsx global>{`
          .delete {
            width: 10px;
            text-align: center;
          }
        `}</style>
      </table>
    )
  }

  return <div>No notes</div>
}
