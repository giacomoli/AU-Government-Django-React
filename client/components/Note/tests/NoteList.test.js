import React from 'react'
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo';
import { reducer } from 'redux-form'
import { ApolloClient } from 'apollo-client';
import { createStore } from 'redux';

import NoteList from '../NoteList'

/* eslint-disable */
describe('Note', () => {

  const client = new ApolloClient({})
  const store = createStore(reducer)

  it('should render Note with full props', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <NoteList
          matterId={123123}
          contactId={12312}
          user={{
            fullName: 'Test Test',
          }}
          notes={[
            0: {
              __typename: "NoteType",
              id: '1',
              key: '1',
              dateTime: '2018-10-02T14:30:00+00:00',
              text: 'Test note',
              user: {
                fullName: 'Admin Test',
              },
            },
            1: {
              __typename: "NoteType",
              id: '2',
              key: '2',
              dateTime: '2018-11-02T14:30:00+00:00',
              text: 'Test second note',
              user: {
                fullName: 'Admin Test',
              },
            },
          ]}
        />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Note without data', () => {
    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <NoteList
          matterId={123123}
          contactId={12312}
          user={{
            fullName: 'Test Test',
          }}
          notes={[]}
        />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
