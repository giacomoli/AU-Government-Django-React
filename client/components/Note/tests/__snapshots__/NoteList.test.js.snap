// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note should render Note with full props 1`] = `
<table
  className="jsx-3305124439 table table-bordered table-striped m-b-0"
  id="mainTable"
>
  <thead
    className="jsx-3305124439"
  >
    <tr
      className="jsx-3305124439"
    >
      <th
        className="jsx-3305124439"
      >
        Date time
      </th>
      <th
        className="jsx-3305124439"
        width="70%"
      >
        Note
      </th>
      <th
        className="jsx-3305124439"
      >
        Staff member
      </th>
    </tr>
  </thead>
  <tbody
    className="jsx-3305124439"
  >
    <tr
      className="jsx-2376464361"
    >
      <td
        className="jsx-2376464361"
        style={
          Object {
            "textAlign": "center",
            "width": 200,
          }
        }
      >
        <div
          className="col"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          Jan 16th 2019
        </div>
      </td>
      <td
        className="jsx-2376464361"
      >
        <div
          className="jsx-2376464361 row"
        >
          <div
            className="col"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
      </td>
      <td
        className="jsx-2376464361"
      />
    </tr>
    <tr
      className="jsx-2376464361"
    >
      <td
        className="jsx-2376464361"
        style={
          Object {
            "textAlign": "center",
            "width": 200,
          }
        }
      >
        <div
          className="col"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          Jan 16th 2019
        </div>
      </td>
      <td
        className="jsx-2376464361"
      >
        <div
          className="jsx-2376464361 row"
        >
          <div
            className="col"
            onClick={[Function]}
            role="button"
            tabIndex={0}
          />
        </div>
      </td>
      <td
        className="jsx-2376464361"
      />
    </tr>
  </tbody>
</table>
`;

exports[`Note should render Note without data 1`] = `
<div>
  No notes
</div>
`;
